odoo/tools/translate.py

```diff
@@ -443,8 +443,23 @@ def _get_lang(self, frame):
                     lang = env['res.users'].context_get()['lang']
         return lang
 
-    def __call__(self, source):
-        return self._get_translation(source)
+    def __call__(self, source, *args, **kw):
+        translation = self._get_translation(source)
+        if args:
+            try:
+                return translation % args
+            except (TypeError, ValueError):
+                bad = translation
+                translation = source % args
+                _logger.error('Bad translation "%r" for "%r"', bad, source)
```

Comment:
Might be useful to use `logging.exception` or `exc_info=True` here. And some of the errors are related to incorrect values rather than an incorrect translation string.

Also `%r` does the quoting, so they're not necessary.

Reply:
Would it be useful to add the language in the error? (need a new call to `_get_lang`)

Reply:
Maybe. Would that actually help? E.g. let's say you see such an error in the log, how would it get fixed, would you go to the translation list and look for the proper source / improper translation, or would you first filter by language then look for the proper source / improper translation?

Reply:
There is a good chance the admin knows what language are installed (except on the SaaS were you handle tons of dbs). And if not sure, can still make a global search on the db.

Reply:
Either way it probably never hurts to have e.g. "Bad <lang> translation %r for %r" so might as well put it in if it's not too complicated.

Reply:
Though maybe this could be done by modifying _get_translation to return both the translation and the lang, or delegate to a method which does that.